### PR TITLE
Camera controller as React component

### PR DESCRIPTION
**Pull request by philippotto**
### Steps to test:
- Usual testing with special focus on switching between the modes as well as all camera related things in plane mode (zoom, rotate in 3D view etc).

### Issues:
- #1858

------
- [X] Ready for review

### CODE REVIEW

**Comment by philippotto** [comment]
We should probably also test the changes with big tracings to find out whether there is a performance hit.

File: app/assets/javascripts/oxalis/controller/viewmodes/plane_controller.js [hunk 1]
```diff
@@ -26,16 +26,25 @@ import PlaneView from "oxalis/view/plane_view";
...
+  onRender: () => void,
+};
+
+type Props = OwnProps & {
+  flycam: FlycamType,
+  scale: Vector3,
+};
 
 class PlaneController extends React.PureComponent {
   // See comment in Controller class on general controller architecture.
```
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/controller/viewmodes/plane_controller.js]: Is this comment still valid?

File: app/assets/javascripts/oxalis/controller/viewmodes/plane_controller.js [hunk 2]
```diff
@@ -137,39 +182,19 @@ class PlaneController extends React.PureComponent {
...
-      this.cameraController.changeTDViewXY,
-      this.cameraController.changeTDViewYZ,
-      this.cameraController.changeTDViewXZ,
+      api.tracing.changeTDViewDiagonal,
+      api.tracing.changeTDViewXY,
+      api.tracing.changeTDViewYZ,
+      api.tracing.changeTDViewXZ,
     ];
 
     $("#TDViewControls button")
```
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/controller/viewmodes/plane_controller.js]: Can you remove Jquery here and use vanilla JS, please.

File: app/assets/javascripts/oxalis/controller/viewmodes/plane_controller.js (hunk 1 above)
Comment (philippotto) [inline app/assets/javascripts/oxalis/controller/viewmodes/plane_controller.js]: Yes, I think so, we still have several controllers for the different tracing modes.

File: app/assets/javascripts/oxalis/api/api_latest.js [hunk 3]
```diff
@@ -288,6 +290,24 @@ class TracingApi {
...
+   * Centers the 3D view.
+   *
+   * @example
+   * api.tracing.centerTDView()
+   */
+  centerTDView = (): void => {
+    Store.dispatch(centerTDViewAction());
+  }
+
+  changeTDViewXY = (): void => changeTDView(OrthoViews.PLANE_XY);
```
Comment (philippotto) [inline app/assets/javascripts/oxalis/api/api_latest.js]: You're right, I just moved these methods from the camera controller, but the names are definitely not self-explanatory. These methods are used when clicking on the "xy"/... buttons in the 3D view. So, it only changes the 3D view.
How about `rotate3DViewToXY` etc. for the new names? What do you think?

**Comment by philippotto** [comment]
> I tested the PR and it works. However I don't understand all the changes. Perhaps you can give a high level overview of what is happeing here.
>
> Why are some cameras saved in the store and others not?

Sure thing. I refactored the camera controller so that it doesn't have any methods which are used from outside the controller. This mainly involved moving code around (into API and reducers for example). This made it possible to convert the controller to a react component.

One tricky point was that the 3D view camera has to be changed from outside the controller. This is why the data of the 3D camera had to be moved to the store. Unfortunately, this means that we have camera data in the store and the "standard ThreeJS camera". So, we need to sync these two places. A react-component like ThreeJS abstraction could help here, but for now I did it with listenToStore property etc. I think that's alright for now.
Since this was only necessary for the 3D view camera, I didn't do that for the other cameras.

File: app/assets/javascripts/oxalis/api/api_latest.js (hunk 3 above)
Comment (hotzenklotz) [inline app/assets/javascripts/oxalis/api/api_latest.js]: Sure.

File: app/assets/javascripts/oxalis/model/actions/view_mode_actions.js [hunk 4]
```diff
@@ -1,18 +1,76 @@
...
+import * as THREE from "three";
+import { getTDViewportSize } from "oxalis/model/accessors/view_mode_accessor";
 
 type SetViewportActionType = {
   type: "SET_VIEWPORT",
   viewport: OrthoViewType,
 };
 
+type SetTDCameraActionType = {
+	type: "SET_TD_CAMERA",
```
Comment (daniel-wer) [inline app/assets/javascripts/oxalis/model/actions/view_mode_actions.js]: Looks like you mixed tabs and spaces here. We're using spaces everywhere.
This won't happen again if you install the EditorConfig package for Sublime or Atom as we're having a .editorconfig file in the repo :)

Comment (philippotto) [inline app/assets/javascripts/oxalis/model/actions/view_mode_actions.js]: Well spotted! I forgot to install the editor config package after I reinstalled my system. It's done now ;)

**Comment by philippotto** [comment]
@daniel-wer I incorporated the feedback. Can you re-review / accept them? :)

**Pull request merged**